lib/copy-sync/copy-sync.js

```diff
@@ -89,12 +91,24 @@ function copyFileFallback (srcStat, src, dest, opts) {
     pos += bytesRead
   }
 
-  if (opts.preserveTimestamps) fs.futimesSync(fdw, srcStat.atime, srcStat.mtime)
+  setDestTimestampsAndMode(srcStat, src, fdw, opts)
 
   fs.closeSync(fdr)
   fs.closeSync(fdw)
 }
 
+function setDestTimestampsAndMode (srcStat, src, dest, opts) {
+  const utimesSync = typeof dest === 'string' ? utimesMillisSync : fs.futimesSync
+  const chmodSync = typeof dest === 'string' ? fs.chmodSync : fs.fchmodSync
+  if (opts.preserveTimestamps) {
+    // The initial srcStat.atime cannot be trusted because it is modified by the read(2) system call
+    // (See https://nodejs.org/api/fs.html#fs_stat_time_values)
+    const updatedSrcStat = fs.statSync(src)
+    utimesSync(dest, updatedSrcStat.atime, srcStat.mtime)
```

Comment:
I understand that `srcStat.mtime` and `updatedSrcStat.mtime` should be the same, but having it like this looks a little weird! So why not just using the `updatedSrcStat` for both `atime` and `mtime` to make it easier to read?!

Reply:
Fixed in [75756c4](https://github.com/jprichardson/node-fs-extra/pull/633/commits/75756c414d5542cd69789c41ab48dfc6268ae569)